src/check_names.cpp

```diff
@@ -0,0 +1,74 @@
+#include <Rcpp.h>
+using namespace Rcpp;
+
+// jsonify a character vector
+std::string jsonify(CharacterVector x) {
+  std::string out = "[";
+  for (int i = 0; i < x.size(); i++) {
+    out += "\"" + x[i] + "\",";
```

Comment:
Using += is more efficient because it avoids extra copies, so:
out += "\"";
out += x[i];
out += "\",";

would be the preferred way